Use intermediate object to store confirmed matches within a penalty group and prevent infinite reactivation of Inherited virtualization rule (bsc#1094524)

hustodemon opened the pull request:
Previously, the big "accumulate" in calculate12Penalties was too slow
and didn't even finish on some setups.

moio commented on src/main/resources/com/suse/matcher/rules/optaplanner/Scores.drl:
```diff
@@ -23,20 +24,26 @@ global HardSoftScoreHolder scoreHolder;
   then
     // ...add a 50 cent penalty. This ensures a whole number of subscriptions is
     // always used for the same penalty group
+    insertLogical(new ConfirmedMatchInPenaltyGroup($subscriptionId, $penaltyGroupId, $guestId));
```
It seems to me that the code does not agree with the comment any more, please reword the latter to better explain what you are doing, thanks!

hustodemon commented:
Fixed and added one more commit that prevents infinite addition of matches based on inherited virtualization group (the original code allowed inserting `PartialMatches` based on IV subscriptions and `PartialMatches` that also were generated this way).
